jmx-metrics/src/test/groovy/io/opentelemetry/contrib/jmxmetrics/IntegrationTest.groovy

```diff
@@ -1,257 +0,0 @@
-                MountableFile.forHostPath(scriptPath), "/app/${scriptName}")
-                .withCopyFileToContainer(
-                MountableFile.forHostPath(configPath), "/app/${configName}")
-                .withCommand(jvmCommand as String[])
-                .withStartupTimeout(Duration.ofSeconds(120))
-                .waitingFor(Wait.forLogMessage(".*Started GroovyRunner.*", 1))
-                .dependsOn(targetContainers)
-        if (prometheusPort != 0) {
-            jmxExtensionAppContainer.withExposedPorts(prometheusPort)
-        }
-        jmxExtensionAppContainer.start()
-
-        targetContainers.each {
-            assertTrue(it.running)
-        }
-        assertTrue(jmxExtensionAppContainer.running)
-
-        if (prometheusPort != 0) {
-            jmxExposedPort = jmxExtensionAppContainer.getMappedPort(prometheusPort)
-        }
-    }
-}
-
-class OtlpIntegrationTest extends IntegrationTest  {
-
-    @Shared
-    def collector
-    @Shared
-    def collectorServer
-    @Shared
-    def otlpPort
-
-    def setup() {
-        // set up a collector per test to avoid noisy neighbor
-        otlpPort = availablePort()
-        collector = new Collector()
-        collectorServer = ServerBuilder.forPort(otlpPort).addService(collector).build()
-        collectorServer.start()
-    }
-
-    def cleanup() {
-        collectorServer.shutdownNow()
-        collectorServer.awaitTermination(5, TimeUnit.SECONDS)
-    }
-
-    static def availablePort() {
-        def sock = new ServerSocket(0);
-        def port = sock.getLocalPort()
-        sock.close()
-        return port
-    }
-
-    static final class Collector extends MetricsServiceGrpc.MetricsServiceImplBase {
-        private final List<ResourceMetrics> receivedMetrics = new ArrayList<>()
-        private final Object monitor = new Object()
-
-        @Override
-        void export(
-                ExportMetricsServiceRequest request,
-                StreamObserver<ExportMetricsServiceResponse> responseObserver) {
-            synchronized (receivedMetrics) {
-                receivedMetrics.addAll(request.resourceMetricsList)
-            }
-            synchronized (monitor) {
-                monitor.notify()
-            }
-            responseObserver.onNext(ExportMetricsServiceResponse.newBuilder().build())
-            responseObserver.onCompleted()
-        }
-
-        List<ResourceMetrics> getReceivedMetrics() {
-            List<ResourceMetrics> received
-            try {
-                synchronized (monitor) {
-                    monitor.wait(15000)
-                }
-            } catch (final InterruptedException e) {
-                assertTrue(e.message, false)
-            }
-
-            synchronized (receivedMetrics) {
-                received = new ArrayList<>(receivedMetrics)
-                receivedMetrics.clear()
-            }
-            return received
-        }
-    }
-
-    static final String expectedMeterVersion() {
```

Comment:
We appear to be losing instrumentation library version checks with these changes.  Would it be worth adding it to `waitAndAssertMetrics` or an assertion helper?

Reply:
Thanks for the reminder, I should have done it sooner than trying to save to the end 😂